backend/apps/ifc_validation/checks/header_policy/validate_header.py

```diff
@@ -128,19 +165,69 @@ def populate_header(cls, values):
 
     @field_validator('description')
     def validate_description(cls, v, values):
-        # https://github.com/buildingSMART/IFC4.x-IF/tree/header-policy/docs/IFC-file-header#description
-        v = v[0].replace(' ', '') # allow whitespaces
-        allowed_descriptions = yaml.safe_load(open(os.path.join(os.path.dirname(__file__), 'valid_descriptions.yaml')))
-        
-        schema_identifier = values.data.get('file').schema_identifier
+        """
+        https://github.com/buildingSMART/IFC4.x-IF/tree/header-policy/docs/IFC-file-header#description
+        For grammar refer to https://standards.buildingsmart.org/documents/Implementation/ImplementationGuide_IFCHeaderData_Version_1.0.2.pdf
+        """
+        header_description_text = ' '.join(v)
         try:
-            if not v in list(map(lambda desc: desc.replace(" ", ""), allowed_descriptions.get(schema_identifier))):
-                values.data.get('validation_errors').append(values.field_name)
-            if not v: # description field is mandatory
-                values.data.get('validation_errors').append(values.field_name)
+            parsed_description = parse_mvd(header_description_text)
         except:
-            pass  ## ---> what to do with invalid schema identifier? (IFC4X3)
-        return (v,)
+            pass
```

Comment:
Never generic try-except, but also this doesn't work:

If `parsed_description = parse_mvd(header_description_text)` fails then `parsed_description` is undefined, but it's being referenced the line below.

Reply:
Not applicable anymore